pkg/notes/adaptor/repository/prisma.ts

```diff
@@ -0,0 +1,235 @@
+import { Option, Result } from '@mikuroxina/mini-fn';
+import { type Prisma, type PrismaClient } from '@prisma/client';
+
+import type { AccountID } from '../../../accounts/model/account.js';
+import type { prismaClient } from '../../../adaptors/prisma.js';
+import type { ID } from '../../../id/type.js';
+import { Bookmark } from '../../model/bookmark.js';
+import { Note, type NoteID, type NoteVisibility } from '../../model/note.js';
+import type {
+  BookmarkRepository,
+  NoteRepository,
+} from '../../model/repository.js';
+
+type DeserializeNoteArgs = Prisma.PromiseReturnType<
+  typeof prismaClient.note.findUnique
+>;
+
+export class PrismaNoteRepository implements NoteRepository {
+  constructor(private readonly client: PrismaClient) {}
+
+  private serialize(note: Note) {
+    const visibility = () => {
+      switch (note.getVisibility()) {
+        case 'PUBLIC':
+          return 0;
+        case 'HOME':
+          return 1;
+        case 'FOLLOWERS':
+          return 2;
+        case 'DIRECT':
+          return 3;
+      }
+    };
+
+    return {
+      id: note.getID() as string,
+      text: note.getContent(),
+      visibility: visibility(),
+      authorId: note.getAuthorID() as string,
+      createdAt: note.getCreatedAt(),
+      deletedAt: Option.isNone(note.getDeletedAt())
+        ? undefined
+        : Option.unwrap(note.getDeletedAt()),
+    };
+  }
+
+  private deserialize(data: DeserializeNoteArgs): Note {
+    if (!data) throw new Error('Invalid Note data');
```

Comment:
```suggestion
    if (!data) {
      throw new Error('Invalid Note data');
    }
```
コーディングスタイルのガイドを当たったところ明示されてはいませんでしたが, 常に分岐先のブロックを作ってほしいです.